mozdownload/scraper.py

```diff
@@ -870,13 +870,126 @@ def platform_regex(self):
+                        'mac64': r'.*\.%(EXT)s$',
+                        'win32': r'.*(\.installer%(STUB)s)\.%(EXT)s$',
+                        'win64': r'.*(\.installer%(STUB)s)\.%(EXT)s$'}
+
+        regex = regex_base_name + regex_suffix[self.platform]
+
+        return regex % {'APP': self.application,
+                        'LOCALE': self.locale,
+                        'PLATFORM': PLATFORM_FRAGMENTS[self.platform],
+                        'STUB': '-stub' if self.is_stub_installer else '',
+                        'EXT': self.extension}
+
+    def build_filename(self, binary):
+        """Return the proposed filename with extension for the binary"""
+
+        return '%(CHANGESET)s%(DEBUG)s-%(NAME)s' % {
+            'CHANGESET': self.changeset,
+            'DEBUG': '-debug' if self.debug_build else '',
+            'NAME': binary}
+
+    @property
+    def build_list_regex(self):
+        """Return the regex for the folder which contains the list of builds"""
+
+        return 'try-builds'
+
+    def detect_platform(self):
+        """Detect the current platform"""
+
+        platform = Scraper.detect_platform(self)
+
+        # On OS X we have to special case the platform detection code and
+        # fallback to 64 bit builds for the en-US locale
+        if mozinfo.os == 'mac' and self.locale == 'en-US' and \
+                mozinfo.bits == 64:
+            platform = "%s%d" % (mozinfo.os, mozinfo.bits)
+
+        return platform
+
+    def get_build_info_for_index(self, build_index=None):
+        url = urljoin(self.base_url, self.build_list_regex)
+
+        self.logger.info('Retrieving list of builds from %s' % url)
+        parser = DirectoryParser(url, authentication=self.authentication,
+                                 timeout=self.timeout_network)
+        parser.entries = parser.filter('.*-%s$' % self.changeset)
+
+        if not parser.entries:
+            message = 'No builds have been found'
+            raise NotFoundError(message, url)
+
+        self.show_matching_builds(parser.entries)
+
+        # If no index has been given, set it to the last build of the day.
+        if build_index is None:
+            build_index = len(parser.entries) - 1
+        self.logger.info('Selected build: %s' % parser.entries[build_index])
+
+        return (parser.entries, build_index)
+
+    @property
+    def path_regex(self):
+        """Return the regex for the path"""
+
+        try_dir = 'try-%(PLATFORM)s%(DEBUG)s' % {
+            'PLATFORM': self.platform_regex,
+            'DEBUG': '-debug' if self.debug_build else ''}
+        return urljoin(self.build_list_regex,
+                       self.builds[self.build_index],
+                       try_dir)
+
+    @property
+    def platform_regex(self):
+        """Return the platform fragment of the URL"""
+
+        PLATFORM_FRAGMENTS = {'linux': 'linux',
+                              'linux64': 'linux64',
+                              'mac': 'macosx',
```

Comment:
> |detect_platform| does some special casing for this, I believe that
> should be enough. Although maybe that's changed?

That is only called in case of no platform has been specified via the -p
option. So if you explicitly pass in -p mac it would not work.